src/util/getNextThoughtsWithContextChain.js

```diff
@@ -0,0 +1,178 @@
+  // const allowSingleContext = false
+  if (!context.length) {
+    return false
+  }
+  return isContextViewActive(context) && getContextsSortedAndRanked(head(pathToContext(context))).length > 1
+}
+
+const getContextSiblings = (context, thoughtIndex) => {
+  const contexts = getContextsSortedAndRanked(head(context))
+  return mapContextsToThoughts(contexts, thoughtIndex)
+}
+
+const nextSiblingContext = (value, rank, context, thoughtIndex) => {
+  const contextSiblings = getContextSiblings(context, thoughtIndex)
+  const currentIndex = contextSiblings.findIndex(thoughts => thoughts.find(thought => thought.value === value))
+  return contextSiblings[currentIndex + 1]
+}
+
+const firstChildOfContext = (value, cursor, thoughtIndex) => {
+  const context = pathToContext(cursor)
+  const contextSiblings = getContextSiblings(context, thoughtIndex)
+  return contextSiblings[0]
+}
+
+const firstChildOfThought = (thoughtsRanked, showHiddenThoughts) => {
+  const contextMeta = meta(thoughtsRanked)
+  const sortPreference = getSortPreference(contextMeta)
+  const children = (sortPreference === 'Alphabetical' ? getThoughtsSorted : getThoughtsRanked)(thoughtsRanked)
+  const notHidden = child => !isFunction(child.value) && !meta(pathToContext(thoughtsRanked).concat(child.value)).hidden
+  const childrenFiltered = showHiddenThoughts ? children : children.filter(notHidden)
+  return childrenFiltered[0]
+}
+
+const getDerivedCursor = contextChain => {
+  // last of second last item in context chain gives us the current context
+  const context = head(contextChain.slice(-2, -1)[0])
+  const contexts = getContextsSortedAndRanked(context.value)
+  const currentContextTop = head(contextChain)[0].value
+  const matchedContextChain = contexts.find(c => c.context.includes(currentContextTop))
+  const matchedContext = matchedContextChain.context.map(i => ({ value: i, rank: matchedContextChain.rank }))
+
+  return [...matchedContext, context, ...head(contextChain).slice(1)]
+}
+
+const contextViewLookup = (value, context, rank, cursor, contextRanked, contextChain, ignoreChildren) => {
+  const { showHiddenThoughts, thoughtIndex } = store.getState()
+
+  if (context.length === 0 || cursor.length === 0) {
+    return null
+  }
+
+  // if the cursor is at a thought with context view open, move it into context view - jump in
+  if (isValidContextView(pathToContext(cursor))) {
+    const currentThought = head(cursor)
+    const firstChild = firstChildOfContext(value, cursor, thoughtIndex)
+    if (!ignoreChildren && firstChild) {
+      return { nextThoughts: [...firstChild, currentThought], contextChain }
+    }
+    // jump out if there are no context children
+    return getNextThoughtsWithContextChain(value, context, rank, cursor, contextOf(contextRanked), contextOf(contextChain))
+  }
+  // if cursor is within a context view
+  else if (isValidContextView(contextRanked)) {
+    const headContext = contextOf(cursor).map(c => c.value)
```

Comment:
Use `pathToContext` instead of `.map(c => c.value)`.

Reply:
Thanks for sharing. It's updated now.